atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -275,6 +276,8 @@
         this.sweepQueue = sweepQueue;
         this.deleteExecutor = deleteExecutor;
         this.hasReads = false;
+        this.shouldProfile = shouldProfile;
+        this.profileProcessor = createDefaultCommitProfileProcessor();
```

Comment:
Or a spare 5 minutes?
1. pass `shouldProfile` into `createDefaultCommitProfileProcessor`, and remove the field
2. move the `createDefaultCommitProfileProcessor` logic to `CommitProfileProcessor`, taking `getTimer` and `getHistogram` with it. All the calls to those methods would become `profileProcessor.getTimer`, which doesn't sound too bad. I would keep `SnapshotTransaction.class` as the metric class in `getTimer` and `getHistogram` for legacy reasons, and because the `ProfileProcessor` is responsible for recording metrics for SnapshotTransactions.

Reply:
There are some non-commit things that also use `getTimer`/`getHistogram` though that aren't part of commit-stage profiling (e.g. `getRows`), so I wouldn't want to exclusively move these into the profiler.

We can discuss ideas for cleaning this up offline. The tricky bit here is that serializable transactions create read transactions of their own (we could of course have 2 constructors where one takes `shouldProfile` and one takes the processor, but that's pretty nasty)